### PR TITLE
Add tests for strict mode

### PR DESCRIPTION
**Pull request by sadym-chromium**
Add test for strict mode in `evaluate` and `callFunction`

### CODE REVIEW

File: tests/_helpers.py [hunk 1]
```diff
@@ -110,16 +110,33 @@ def recursive_compare(expected, actual):
...
+        f"but is {type(actual)} instead."
+
+
+def any_int(actual):
+    assert isinstance(actual, int), \
+        f"'{actual}' should be string, " \
+        f"but is {type(actual)} instead."
+
+
+def any_timestamp(actual):
```
Comment (thiagowfx) [inline tests/_helpers.py:136]: How about `any_number` instead?

**Review by thiagowfx — APPROVED** [review]

**Comment by mathiasbynens** [comment]
Why are these not WPT tests?

**Comment by sadym-chromium** [comment]
> Why are these not WPT tests?

There are such tests in WPT as well: https://github.com/web-platform-tests/wpt/pull/35448

Some of our e2e tests intersect with WPT, and this is an example. Do you think this tests are too niche, and should not be in e2e tests? I'm inclined to agree with that position.

**Comment by sadym-chromium** [comment]
The tests are too niche, and we can keep them only in WPT: https://github.com/web-platform-tests/wpt/pull/35448

**Pull request closed without merging**